operator/src/main/java/org/projectnessie/operator/resource/NessieSpec.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+public class NessieSpec {
+
+  public enum VersionStoreType {
+    IN_MEMORY,
+    ROCKSDB,
+    DYNAMODB,
+    MONGODB,
+    CASSANDRA,
+    JDBC,
+    BIGTABLE
+  }
+
+  @JsonPropertyDescription("The number of replicas to run.")
+  private int size = 1;
+
+  @JsonPropertyDescription("The image to use for the Nessie server.")
+  private NessieImage image = new NessieImage();
+
+  @JsonPropertyDescription("The log level to use for the Nessie server.")
+  private String logLevel = "INFO";
+
+  @JsonPropertyDescription("The type of version store to use.")
+  private VersionStoreType versionStoreType = VersionStoreType.IN_MEMORY;
```

Comment:
I wonder whether the operator should allow `IN_MEMORY` only with some `reallyUseInMemoryBecauseIAmAwareThatThingsAreNotPersisted` flag (maybe not that long). WDYT?

Reply:
Is it useful for the operation to support `IN_MEMORY` at all? A single Nessie pod can probably be managed with a single k8s yaml file.

Reply:
Imho useful for testing / prototyping. We can add the flags later on, for now my goal is to achieve feature parity with the Helm chart.

Reply:
Can we fail with a clear error message if there are more than 1 `IN_MEMORY` Nessie pod?